src/nvim/normal.c

```diff
@@ -1468,6 +1468,32 @@ static void move_tab_to_mouse(void)
   }
 }
 
+/// Check that mouse support is enabled for visual mode
+static bool mouse_supports_visual(void)
```

Comment:
this is rather specific and looks mostly copy-pasted from `ui_check_mouse`. Any way to extract the logic from `ui_check_mouse` into something like `mouse_is_mode_enabled` ?

Reply:
Yes, and previous code was seperating this logic in a separated function, but the refactoring that happened in 6db86cb2d3d4ca152f156dc07362f8796150fae0 moved it (since only one call was needed).

Reply:
https://github.com/neovim/neovim/commit/6db86cb2d3d4ca152f156dc07362f8796150fae0 inlined `mouse_has()` because there didn't seem to be a need for it. If there is now a need for it (or something like it), then it can be restored... that seems preferable to copy-pasting logic.